pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1231,7 +1231,16 @@ protected OffloadProcessStatus internalOffloadStatus(boolean authoritative) {
      */
     private Topic getTopicReference(TopicName topicName) {
         return pulsar().getBrokerService().getTopicIfExists(topicName.toString()).join()
-                .orElseThrow(() -> new RestException(Status.NOT_FOUND, "Topic not found"));
+                .orElseThrow(() -> {
+                    if (topicName.getCompleteTopicName().contains(TopicName.PARTITIONED_TOPIC_SUFFIX)) {
+                        if (!internalGetPartitionedTopicList().contains(topicName.getPartitionedTopicName())) {
```

Comment:
I think `internalGetPartitionedTopicList()` is not appropriate here because it will try to get the list of partitioned topics for a specific namespace. Rather, we should just do a point check for this specific `topicName` and check the partitioned topic metadata for the validation.